Fluid conduit crashes game if it has non-existant fluid

jollycar opened the pull request:
# Description

Fluid conduit crashes game if it has non-existant fluid

If a fluid is removed from the game, minecraft / forge will assign 
the fluid to the instance "Fluids.EMPTY".

The nbt fluid string will still be kept (to the non-existant fluid) 
initially, but if you lookup the fluid string in the registries, 
it will return "Fluids.EMPTY".

The fix is to set the lockedFluid to null on deserialization, 
if the fluid, after lookup, is equal to "Fluids.EMPTY".

closes #382

justliliandev commented:
instead of all those changes, can you rather check on deserialization that the fluid is not the empty fluid and if it is, set it to null

jollycar committed:
Fluid conduit crashes game if it has non-existant fluid

If a fluid is removed from the game, minecraft / forge will assign
the fluid to the instance "Fluids.EMPTY".

The nbt fluid string will still be kept (to the non-existant fluid)
initially, but if you lookup the fluid string in the registries,
it will return "Fluids.EMPTY".

The fix is to set the lockedFluid to null on deserialization,
if the fluid, after lookup, is equal to "Fluids.EMPTY".

closes #382

jollycar commented:
good call. see new code change (I rebased and force pushed)

justliliandev commented:
the serialized fluidid is not `minecraft:empty`. you would have to query the Registry for it's value and if it's empty, set the locked fluid to null

jollycar commented:
> the serialized fluidid is not `minecraft:empty`. you would have to query the Registry for it's value and if it's empty, set the locked fluid to null

The value for `Fluids.EMPTY` that is being deserialized **IS** `minecraft:empty`. It is not empty (assuming you mean an empty string) Try printing
```java
ForgeRegistries.FLUIDS.getKey(Fluids.EMPTY).toString()
```

Remember `Fluids.EMPTY` is the fluid that is in the conduit when the previous fluid is removed from the game. This is done by forge.

I don't understand exactly what you're suggesting. Should I put `ForgeRegistries.FLUIDS.getKey(Fluids.EMPTY).toString()` instead of `minecraft:empty`? And what do you mean by if it's empty. It's not empty.

thedarkcolour commented on src/conduits/java/com/enderio/conduits/common/types/FluidExtendedData.java:
```diff
@@ -70,7 +70,7 @@ public CompoundTag serializeGuiNBT() {
         if (nbt.contains(ConduitNBTKeys.FLUID) && !isMultiFluid) {
             String fluid = nbt.getString(ConduitNBTKeys.FLUID);
-            if (fluid.equals("null")) {
+            if (fluid.equals("null") || fluid.equals("minecraft:empty")) {
```
There is no guarantee that the string will be "minecraft:empty" (it could just be "empty") so I would use something like `new ResourceLocation(fluid).toString().equals("minecraft:empty")`

justliliandev commented:
The fluid that will be serialized could be "astralsorcery:liquid_starlight", when retrieving that string it will still be "astralsorcery:liquid_starlight" even though astral sorcery has been removed, therefore this string comparison will fail. just deserialize the fluid as normal, but if it is the empty fluid, set it to null

jollycar commented:
rebased and force pushed
